src/InstrumentationEngine.Lib/RawProfilerHookSettingsReader.cpp

```diff
@@ -27,45 +33,74 @@ namespace MicrosoftInstrumentationEngine
 
         std::wstring strRawProfilerHookClsid;
 
-        CLogging::LogMessage(_T("Examining environment variable: %s"), cszRawProfilerHookVariableName);
+        // If CoreCLR RPH variable is not set, fallback to the desktop one if it exists.
+        // We only want to use the CoreCLR RPH path variable if the CoreCLR classId was set.
+        bool fUseCoreClrPath; 
+        if (runtimeType == COR_PRF_RUNTIME_TYPE::COR_PRF_CORE_CLR)
+        {
+            fUseCoreClrPath = true;
+            CLogging::LogMessage(_T("Examining coreCLR environment variable: %s"), cszCoreRawProfilerHookVarName);
+            IfFailRet(GetEnvironmentVariableWrapper(cszCoreRawProfilerHookVarName, strRawProfilerHookClsid));
```

Comment:
Should we be logging if the environment variable isn't found here?

Reply:
Sure, that will help with tracking what happened here.

---
In reply to: [436912916](https://github.com/microsoft/CLRInstrumentationEngine/pull/270#discussion_r436912916) [](ancestors = 436912916)

Reply:
Yes, good idea